cdc/capture/http_handler.go

```diff
@@ -0,0 +1,620 @@
+// @Accept  json
+// @Produce  json
+// @Success 200 {array} model.ProcessorCommonInfo
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/processors [get]
+func (h *httpHandler) ListProcessor(c *gin.Context) {
+	infos, err := h.capture.etcdClient.GetProcessors(c)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+	resps := make([]*model.ProcessorCommonInfo, len(infos))
+	for i, info := range infos {
+		resp := &model.ProcessorCommonInfo{CfID: info.CfID, CaptureID: info.CaptureID}
+		resps[i] = resp
+	}
+	c.IndentedJSON(http.StatusOK, resps)
+}
+
+// ListCapture lists all captures
+// @Summary List captures
+// @Description list all captures in cdc cluster
+// @Tags capture
+// @Accept  json
+// @Produce  json
+// @Success 200 {array} model.Capture
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/captures [get]
+func (h *httpHandler) ListCapture(c *gin.Context) {
+	_, raw, err := h.capture.etcdClient.GetCaptures(c)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+
+	ownerID, err := h.capture.etcdClient.GetOwnerID(c, kv.CaptureOwnerKey)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+
+	captures := make([]*model.Capture, 0, len(raw))
+	for _, c := range raw {
+		isOwner := c.ID == ownerID
+		captures = append(captures,
+			&model.Capture{ID: c.ID, IsOwner: isOwner, AdvertiseAddr: c.AdvertiseAddr})
+	}
+
+	c.IndentedJSON(http.StatusOK, captures)
+}
+
+// ServerStatus gets the status of server(capture)
+// @Summary Get server status
+// @Description get the status of a server(capture)
+// @Tags common
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} model.ServerStatus
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/status [get]
+func (h *httpHandler) ServerStatus(c *gin.Context) {
+	status := model.ServerStatus{
+		Version: version.ReleaseVersion,
+		GitHash: version.GitHash,
+		Pid:     os.Getpid(),
+	}
+	status.ID = h.capture.Info().ID
+	status.IsOwner = h.capture.IsOwner()
+	c.IndentedJSON(http.StatusOK, status)
+}
+
+// Health check if cdc cluster is health
+// @Summary Check if CDC cluster is health
+// @Description check if CDC cluster is health
+// @Tags common
+// @Accept  json
+// @Produce  json
+// @Success 200
+// @Failure 500 {object} model.HTTPError
+// @Router	/api/v1/health [get]
+func (h *httpHandler) Health(c *gin.Context) {
+	if _, err := h.capture.GetOwner(c); err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
+		return
+	}
+	c.Status(http.StatusOK)
+}
+
+// forwardToOwner an request to owner
+func (h *httpHandler) forwardToOwner(c *gin.Context) {
+	// every request can only forward to owner one time
+	if len(c.GetHeader(ForwardFromCapture)) != 0 {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(cerror.ErrOwnerNotFound))
```

Comment:
Better to use another error, this error is confused

Reply:
done